coordinator/src/position/trade_position.rs

```diff
@@ -0,0 +1,432 @@
+        .await
+        .expect("task to complete")?;
+
+    let contract_states = contracts
+        .iter()
+        .map(|contract| (contract.get_temporary_id(), ContractState::from(contract)))
+        .collect();
+
+    let current_contracts = compute_trade_position_maker(trades, contract_states);
+
+    for contract_symbol in all::<ContractSymbol>() {
+        let previous_contracts = previous_contracts.get(&contract_symbol);
+        let current_contracts = current_contracts.get(&contract_symbol);
+        if previous_contracts != current_contracts {
+            tracing::debug!(
+                %contract_symbol,
+                contracts_before = %previous_contracts,
+                contracts_after = %current_contracts,
+                "Updating maker simulated position"
+            );
+
+            let res = spawn_blocking({
+                let maker_endpoint = maker_endpoint.to_string();
+                move || {
+                    let contract_symbol = match contract_symbol {
+                        ContractSymbol::BtcUsd => maker::routes::ContractSymbol::BtcUsd,
+                    };
+
+                    let maker_endpoint = maker_update_position_endpoint(&maker_endpoint);
+
+                    ureq::put(&maker_endpoint).send_json(maker::routes::UpdatePositionRequest {
+                        contract_symbol,
+                        contracts: current_contracts,
+                    })
+                }
+            })
+            .await
+            .expect("task to complete")?;
+
+            ensure!(
+                res.status() == 200,
+                "Could not update maker simulated position"
+            );
+
+            tracing::info!(
+                %contract_symbol,
+                contracts_before = %previous_contracts,
+                contracts_after = %current_contracts,
+                "Updated maker simulated position"
+            );
+        }
+    }
+
+    *previous_contracts = current_contracts;
+
+    Ok(())
+}
+
+/// Computes the trade position of the _maker_.
+///
+/// The maker's trade position is calculated based on the coordinator's combined position with all
+/// app users. No transformation is needed after calculating the coordinator's position. To
+/// understand why, consider this example:
+///
+/// - 1 (total) app user:               10 long.
+/// - Coordinator with apps:            10 short.
+/// - Simulated coordinator with maker: 10 long.
+/// - Simulated maker with coordinator: 10 short.
+///
+/// Basically, coordinator-apps (total) is equal to maker-coordinator.
+///
+/// It's important to note that this is the position _caused by trading_. That is, it
+/// excludes force-closures, position expiries, etc.
+fn compute_trade_position_maker(
+    trades: Vec<TradeForContract>,
+    contract_states: HashMap<[u8; 32], ContractState>,
+) -> ContractsPerContractSymbol {
+    let grouped_trades = &trades
```

Comment:
❓ How will this logic behave when we are using rollovers, which is also changing the `temporary_contract_id`.